src/MoBi.Presentation/Presenter/SelectObjectInTreePresenter.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections.Generic;
+using Antlr.Runtime.Misc;
+using MoBi.Core.Domain.Model;
+using MoBi.Presentation.DTO;
+using MoBi.Presentation.Views;
+using OSPSuite.Core.Domain;
+using OSPSuite.Presentation.Nodes;
+using OSPSuite.Presentation.Presenters;
+
+namespace MoBi.Presentation.Presenter
+{
+   public interface ISelectEntityInTreePresenter : IPresenter<ISelectEntityInTreeView>
+   {
+      Func<ObjectBaseDTO, IReadOnlyList<ObjectBaseDTO>> GetChildren { get; set; }
+      bool IsValidSelection(ObjectBaseDTO selectedDTO);
+      void InitTreeStructure(IReadOnlyList<ObjectBaseDTO> entityDTOs);
+      IEntity SelectedEntity { get; }
+      ObjectBaseDTO SelectedDTO { get; }
+      ITreeNode TreeNodeFor(ObjectBaseDTO dto);
+      IObjectBase ObjectFrom(ObjectBaseDTO parentDTO);
+   }
+
+   public class SelectEntityInTreePresenter : AbstractPresenter<ISelectEntityInTreeView, ISelectEntityInTreePresenter>, ISelectEntityInTreePresenter
```

Comment:
that componet is superdumb
does nto even know how to iterate through children but the prop is passed as  afunc